CMake/external_libcurl.cmake

```diff
@@ -0,0 +1,33 @@
+if(ENABLE_RS_AUTO_UPDATER)
+    include(ExternalProject)
```

Comment:
ExternalProject?
Also, why the "external_" in the name of this .cmake? Why not just "libcurl.cmake"?

Reply:
in order to use CMake "ExternalProject_Add" we need to include "ExternalProject"
The ExternalProject_Add() function creates a custom target to drive download, update/patch, configure, build, install and test steps of an external project:

about the name I took libusb as an example which named this way, maybe it was names this way to imply that this is an external project that the LRS consumes.